inabox/tests/payments/reservation_only_test.go

```diff
@@ -0,0 +1,234 @@
+	"github.com/Layr-Labs/eigenda/test"
+	"github.com/Layr-Labs/eigenda/test/random"
+	"github.com/stretchr/testify/require"
+)
+
+// NOTE: Currently, it doesn't work to run these tests in sequence. Each test must be run as a separate command.
+// The problem is that the cleanup logic sometimes randomly fails to free docker ports, so subsequent setups fail.
+// Once we figure out why resources aren't being freed, then these tests will be runnable the "normal" way.
+
+func TestReservationOnly_LegacyClient_LegacyController(t *testing.T) {
+	testReservationOnly(t, clientledger.ClientLedgerModeLegacy, false)
+}
+
+func TestReservationOnly_LegacyClient_NewController(t *testing.T) {
+	testReservationOnly(t, clientledger.ClientLedgerModeLegacy, true)
+}
+
+func TestReservationOnly_NewClient_LegacyController(t *testing.T) {
+	testReservationOnly(t, clientledger.ClientLedgerModeReservationOnly, false)
+}
+
+func TestReservationOnly_NewClient_NewController(t *testing.T) {
+	testReservationOnly(t, clientledger.ClientLedgerModeReservationOnly, true)
+}
+
+func testReservationOnly(t *testing.T, clientLedgerMode clientledger.ClientLedgerMode, controllerUseNewPayments bool) {
+	// Save current working directory. The setup process in its current form changes working directory, which causes
+	// subsequent executions to fail, since the process relies on relative paths. This is a workaround for now: we just
+	// capture the original working directory, and switch back to it as a cleanup step.
+	originalDir, err := os.Getwd()
+	require.NoError(t, err)
+
+	infraConfig := &integration.InfrastructureConfig{
+		TemplateName:                    "testconfig-anvil.yaml",
+		TestName:                        "",
+		InMemoryBlobStore:               false,
+		Logger:                          test.GetLogger(),
+		RootPath:                        "../../../",
+		UserReservationSymbolsPerSecond: 1024,
+		ClientLedgerMode:                clientLedgerMode,
+		ControllerUseNewPayments:        controllerUseNewPayments,
+	}
+
+	infra, err := integration.SetupInfrastructure(infraConfig)
+	require.NoError(t, err)
+
+	testHarness, err := integration.NewTestHarnessWithSetup(infra)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		testHarness.Cleanup()
+		integration.TeardownInfrastructure(infra)
+
+		if err := os.Chdir(originalDir); err != nil {
+			t.Logf("Failed to restore working directory: %v", err)
+		}
+	})
+
+	integration.MineAnvilBlocks(t, testHarness.RPCClient, 6)
+
+	payloadSize := 1000
+	testDuration := 1 * time.Minute
+	t.Logf("Test Duration: %s", testDuration)
+	t.Logf("Payload size: %d bytes", payloadSize)
+
+	t.Run("Within reservation limits", func(t *testing.T) {
+		initialReservation, err := reservation.NewReservation(
+			1024,
+			time.Now().Add(-1*time.Hour),
+			time.Now().Add(24*time.Hour),
+			[]core.QuorumID{0, 1},
+		)
+		require.NoError(t, err)
+		err = testHarness.UpdateReservation(t.Context(), t, initialReservation)
```

Comment:
nit: can we rename `UpdateReservationOnchain` or something to make it obvious we're making an eth tx? otherwise it could read as "read from onchain and update the test harness' reservation"